src/Control/Distributed/Process/Internal/Primitives.hs

```diff
@@ -982,7 +983,10 @@ expectTimeout n = receiveTimeout n [match return]
 spawnAsync :: NodeId -> Closure (Process ()) -> Process SpawnRef
 spawnAsync nid proc = do
   spawnRef <- getSpawnRef
-  sendCtrlMsg (Just nid) $ Spawn proc spawnRef
+  node     <- getSelfNode
+  if nid == node
+    then sendCtrlMsg Nothing $! Spawn (DS.force proc) spawnRef
```

Comment:
> I found at least one case where deep evaluation of the closure produced an error, and it was an expensive error to find as it crashes the NC.

That's a problem with the definition of `Closure`, not something we should be papering over by forcing evaluation defensively everywhere in the code.